### ISSUE TITLE
Petra Venj items unavailable

### ISSUE THREAD

**Issue by vthornheart-bng** [Bungie-net/api#704]
We've discovered a bug where, if you're not actively playing in the dreaming city with the character making the Vendors request, the vendors API will not return items for Petra Venj.

The reason for this is actually kind of interesting for anyone wondering why this is happening.  We have a single conceptual vendor in content for Petra Venj, which is being used in two different places in the world and in your game progress: once in the Tangled Shore, and once in the Dreaming City.

As a result, she has the same inventory that needs to be made available/unavailable in each.  The folks who built the content for her gated what items should be available, in part, based on what location you're at when you pull for her vendor information: but for the API, that location is often nowhere, or somewhere entirely different.  As a result, those items aren't actually being sold by Petra when you request them: they're only being sold from the moment you touch down at the Dreaming City.

I've filed a request internally to see if this content or these checks can be changed so that we can return this info from the API once more: but until then, we won't be able to return any data about Petra Venj's items for sale unless you're actively playing in the Dreaming City.  Sorry folks!

**Comment by lowlines** [comment]
Oh wow! I can definitely see that in the map data returned from the APIs! Welp! This will make it impossible to determine what week it is by looking at her inventory...

**Comment by vthornheart-bng** [comment]
Yeah, it's a total pain in the butt for all of us, that's for sure.

**Comment by lowlines** [comment]
I just figured out a way to dynamically figure out what week it is in the Dreaming City.
2103539490 - The Strand (Week 1)
2471849886 - Divalian Mists (Week 2)
2101431249 - Rheasilvia (Week 3)

Whichever quest step is in your character’s inventory is the active week and it works on characters that aren’t logged in too. The downside is that this item gets removed from your inventory once completed.

**Comment by vthornheart-bng** [comment]
Ah, very clever!

**Comment by lasdem** [comment]
Any update on resolving this issue?

**Comment by vthornheart-bng** [comment]
I have been informed that the designers will not have time to change the way Petra works in order for us to get correct data.  It would apparently require a significant reworking of how she functions, both in the Dreaming City and during the periods of time where she's in the Tangled Shore.

Unfortunately, this means that Petra will never show data unless you are actively logged in and at the Dreaming City.

I apologize for any inconvenience this may cause.  I've had conversations with the team since this occurred, and everyone is now aware of why this happened: this will hopefully prevent vendors from being set up in such a way in the future.

**Comment by lowlines** [comment]
General rule of thumb, if its not an API issue and it is exclusive to a specific piece of shipped content, the chances of it getting fixed will always be very low to probably never.

TLDR; Petra is cursed.

**Comment by lasdem** [comment]
Since the inventory of Petra Venj is the same for each character and only changes once per day, wouldn't it be possible to check her with a virtual dummy character that is in the correct location and then deliver the correct response to any account/character that queries her via the API?
This would then be a purely API change with the content being untouched. 
Although it would be quite a workaround :-\

**Comment by vthornheart-bng** [comment]
We'd literally need a character to be constantly logged into the game in order to do so, unfortunately: which is something we could neither guarantee nor automate easily.  It'd be a very shaky foundation on which to build the API data.

**Comment by ttgmpsn** [comment]
Does this also mean that Clan Challenges from Tower Hawthorne are never going to be available (see #677)? The case might be a bit different, because one bounty (LW raid one) is actually shown in the API.

Also, I guess you can close #655 now ;)

**Comment by altbdoor** [comment]
Question, is this still an issue, considering how Petra's stuff is visible in API now?

Example: https://www.light.gg/db/vendors/1841717884/petra-venj/

**Comment by justrealmilk** [comment]
> Question, is this still an issue, considering how Petra's stuff is visible in API now?
> 
> Example: https://www.light.gg/db/vendors/1841717884/petra-venj/

that's static manifest data - this is about the Vendor endpoint

**Comment by jshaffstall-bng** [comment]
I think we can finally close this now. Her inventory depended on state from being in the Dreaming City, because she also could be in the Tangled Shore.

But now the Tangled Shore is RIP in pepperoni, so the sale items don't need to check for Dreaming City state.
